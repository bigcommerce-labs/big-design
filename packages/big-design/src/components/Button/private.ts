import { theme as defaultTheme } from '@bigcommerce/big-design-theme';
import styled from 'styled-components';

import { ButtonProps, StyleableButton as _StyleableButton } from './Button';

export const StyleableButton = _StyleableButton;

export const MessagingButton = styled(_StyleableButton)<ButtonProps>`
  background-color: transparent;
  border: ${({ theme }) => theme.border.none};
  color: ${({ theme }) => theme.colors.secondary70};
  height: auto;
  padding: ${({ theme }) => theme.spacing.none};

  &:active {
    background-color: ${({ theme }) => theme.colors.primary20};
  }

  &:focus {
    box-shadow: ${({ theme }) => `0 0 0 ${theme.spacing.xxSmall} ${theme.colors.primary20}`};
  }

  &:hover:not(:active) {
    background-color: ${({ theme }) => theme.colors.primary10};
  }

  &[disabled] {
    color: ${({ theme }) => theme.colors.secondary30};
  }
`;

MessagingButton.defaultProps = { theme: defaultTheme };
