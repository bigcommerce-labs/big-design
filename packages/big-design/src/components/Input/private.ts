export { StyledInput, StyledInputWrapper, type StyledInputWrapperProps } from './styled';
