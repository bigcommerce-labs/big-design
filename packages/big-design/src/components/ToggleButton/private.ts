import { ToggleButtonProps as _ToggleButtonProps } from './ToggleButton';
export { ToggleButton } from './ToggleButton';
export type ToggleButtonProps = _ToggleButtonProps;
