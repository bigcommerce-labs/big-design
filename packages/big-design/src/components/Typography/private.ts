export {
  StyleableText,
  StyleableSmall,
  StyleableH0,
  StyleableH1,
  StyleableH2,
  StyleableH3,
  StyleableH4,
} from './Typography';
