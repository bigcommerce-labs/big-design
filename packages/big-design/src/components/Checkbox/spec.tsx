import { fireEvent, render } from '@testing-library/react';
import 'jest-styled-components';
import React from 'react';

import { Checkbox } from './index';
import { StyleableCheckbox } from './private';

describe('render Checkbox', () => {
  test('checked', () => {
    const { container } = render(<Checkbox label="Checked" checked={true} onChange={() => null} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('unchecked', () => {
    const { container } = render(<Checkbox label="Unchecked" checked={false} onChange={() => null} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('indeterminate', () => {
    const { container } = render(<Checkbox label="Unchecked" isIndeterminate={true} onChange={() => null} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});

test('has correct value for checked', () => {
  const { getByTestId } = render(
    <Checkbox label="Checked" checked={true} onChange={() => null} data-testid="checkbox" />,
  );
  const input = getByTestId('checkbox') as HTMLInputElement;

  expect(input.checked).toBe(true);
});

test('has correct value for unchecked', () => {
  const { getByTestId } = render(
    <Checkbox label="Checked" checked={false} onChange={() => null} data-testid="checkbox" />,
  );
  const input = getByTestId('checkbox') as HTMLInputElement;

  expect(input.checked).toBe(false);
});

test('has correct value for indeterminate', () => {
  const { getByTestId } = render(
    <Checkbox label="Checked" isIndeterminate checked={false} onChange={() => null} data-testid="checkbox" />,
  );
  const input = getByTestId('checkbox') as HTMLInputElement;

  expect(input.checked).toBe(false);
});

test('triggers onChange when clicking the checkbox', () => {
  const onChange = jest.fn();
  const { getByTestId } = render(
    <Checkbox label="Checked" checked={true} onChange={onChange} data-testid="checkbox" />,
  );
  const checkbox = getByTestId('checkbox') as HTMLInputElement;

  fireEvent.click(checkbox);

  expect(onChange).toHaveBeenCalled();
});

test('triggers onChange when clicking styled and text label', () => {
  const onChange = jest.fn();
  const { container } = render(<Checkbox label="Checked" checked={true} onChange={onChange} data-testid="checkbox" />);

  const labels = container.querySelectorAll('label');

  labels.forEach(label => fireEvent.click(label));

  expect(onChange).toHaveBeenCalledTimes(2);
});

test('forwards ref', () => {
  const ref = React.createRef<HTMLInputElement>();

  const { container } = render(<Checkbox ref={ref} label="Checked" data-testid="checkbox" />);
  const input = container.querySelector('input');

  expect(input).toBe(ref.current);
});

test('does not forward styles', () => {
  const { container } = render(<Checkbox label="Checked" className="test" style={{ background: 'red' }} />);

  expect(container.getElementsByClassName('test').length).toBe(0);
  expect(container.firstChild).not.toHaveStyle('background: red');
});

test('private StyleableCheckbox forwards styles', () => {
  const { container } = render(<StyleableCheckbox label="Checked" className="test" style={{ background: 'red' }} />);

  expect(container.getElementsByClassName('test').length).toBe(1);
  expect(container.firstChild).toHaveStyle('background: red');
});

test('displays text greyed out when disabled', () => {
  const { container } = render(<Checkbox disabled label="Checked" className="test" />);

  const labels = container.querySelectorAll('label');

  expect(labels[1]).toHaveStyle('color: #B4BAD1');
});
