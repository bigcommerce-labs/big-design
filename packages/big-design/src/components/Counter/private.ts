import { remCalc, theme as defaultTheme } from '@bigcommerce/big-design-theme';
import styled from 'styled-components';

import { StyledInput, StyledInputWrapper, StyledInputWrapperProps } from '../Input/styled';

export const StyledCounterWrapper = styled(StyledInputWrapper)<StyledInputWrapperProps>`
  padding-left: ${({ theme }) => theme.spacing.xxSmall};
  padding-right: ${({ theme }) => theme.spacing.xxSmall};
  width: ${remCalc(125)};
  max-width: ${remCalc(320)};
`;

export const StyledCounterInput = styled(StyledInput)`
  text-align: center;
  padding: ${({ theme }) => theme.spacing.none};
`;

StyledCounterInput.defaultProps = { theme: defaultTheme };
StyledCounterWrapper.defaultProps = { theme: defaultTheme };
