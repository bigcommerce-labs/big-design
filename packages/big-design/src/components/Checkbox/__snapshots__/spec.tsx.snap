// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Checkbox checked 1`] = `
.c6 {
  color: #313440;
  margin: 0 0 1rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5rem;
}

.c6:last-child {
  margin-bottom: 0;
}

.c7 {
  cursor: pointer;
}

.c5 {
  margin-left: 0.5rem;
}

.c0 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c4 {
  -webkit-transition: all 150ms ease-out;
  transition: all 150ms ease-out;
  -webkit-transition-property: border-color,background,box-shadow,color,opacity;
  transition-property: border-color,background,box-shadow,color,opacity;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #3C64F4;
  box-sizing: border-box;
  border: 1px solid #D9DCE9;
  border-color: #3C64F4;
  border-radius: 0.25rem;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 1.25rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 0.125rem;
  margin-top: 0.125rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1.25rem;
}

.c4:hover {
  border-color: #3C64F4;
}

.c1:focus + .c3 {
  box-shadow: 0 0 0 0.25rem #DBE3FE;
}

.c4 svg {
  opacity: 1;
}

<div
  class="c0"
>
  <input
    aria-checked="true"
    aria-labelledby=":r1:"
    checked=""
    class="c1 c2"
    id=":r0:"
    type="checkbox"
  />
  <label
    aria-hidden="true"
    class="c3 c4"
    for=":r0:"
  >
    <svg
      aria-hidden="true"
      class="brjSIf"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M9 16.17 5.53 12.7a.996.996 0 1 0-1.41 1.41l4.18 4.18c.39.39 1.02.39 1.41 0L20.29 7.71a.996.996 0 1 0-1.41-1.41z"
      />
    </svg>
  </label>
  <div
    class="c5"
  >
    <label
      class="c6 c7"
      for=":r0:"
      id=":r1:"
    >
      Checked
    </label>
  </div>
</div>
`;

exports[`render Checkbox disabled checked 1`] = `
.c6 {
  color: #313440;
  margin: 0 0 1rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5rem;
}

.c6:last-child {
  margin-bottom: 0;
}

.c7 {
  cursor: pointer;
  cursor: not-allowed;
}

.c5 {
  margin-left: 0.5rem;
}

.c0 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c4 {
  -webkit-transition: all 150ms ease-out;
  transition: all 150ms ease-out;
  -webkit-transition-property: border-color,background,box-shadow,color,opacity;
  transition-property: border-color,background,box-shadow,color,opacity;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #3C64F4;
  box-sizing: border-box;
  border: 1px solid #D9DCE9;
  border-color: #3C64F4;
  border-radius: 0.25rem;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 1.25rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 0.125rem;
  margin-top: 0.125rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1.25rem;
  background: #D9DCE9;
  border-color: #D9DCE9;
  cursor: not-allowed;
}

.c1:focus + .c3 {
  box-shadow: 0 0 0 0.25rem #DBE3FE;
}

.c4 svg {
  opacity: 1;
}

<div
  class="c0"
>
  <input
    aria-checked="true"
    aria-labelledby=":rj:"
    checked=""
    class="c1 c2"
    disabled=""
    id=":ri:"
    type="checkbox"
  />
  <label
    aria-hidden="true"
    class="c3 c4"
    disabled=""
    for=":ri:"
  >
    <svg
      aria-hidden="true"
      class="brjSIf"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M9 16.17 5.53 12.7a.996.996 0 1 0-1.41 1.41l4.18 4.18c.39.39 1.02.39 1.41 0L20.29 7.71a.996.996 0 1 0-1.41-1.41z"
      />
    </svg>
  </label>
  <div
    class="c5"
  >
    <label
      aria-hidden="true"
      class="c6 c7"
      disabled=""
      for=":ri:"
      id=":rj:"
    >
      Checked
    </label>
  </div>
</div>
`;

exports[`render Checkbox disabled indeterminate 1`] = `
.c6 {
  color: #313440;
  margin: 0 0 1rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5rem;
}

.c6:last-child {
  margin-bottom: 0;
}

.c7 {
  cursor: pointer;
  cursor: not-allowed;
}

.c5 {
  margin-left: 0.5rem;
}

.c0 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c4 {
  -webkit-transition: all 150ms ease-out;
  transition: all 150ms ease-out;
  -webkit-transition-property: border-color,background,box-shadow,color,opacity;
  transition-property: border-color,background,box-shadow,color,opacity;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #3C64F4;
  box-sizing: border-box;
  border: 1px solid #D9DCE9;
  border-color: #3C64F4;
  border-radius: 0.25rem;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 1.25rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 0.125rem;
  margin-top: 0.125rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1.25rem;
  background: #D9DCE9;
  border-color: #D9DCE9;
  cursor: not-allowed;
}

.c1:focus + .c3 {
  box-shadow: 0 0 0 0.25rem #DBE3FE;
}

.c4 svg {
  opacity: 1;
}

<div
  class="c0"
>
  <input
    aria-labelledby=":rp:"
    class="c1 c2"
    disabled=""
    id=":ro:"
    type="checkbox"
  />
  <label
    aria-hidden="true"
    class="c3 c4"
    disabled=""
    for=":ro:"
  >
    <svg
      aria-hidden="true"
      class="brjSIf"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M18 13H6c-.55 0-1-.45-1-1s.45-1 1-1h12c.55 0 1 .45 1 1s-.45 1-1 1"
      />
    </svg>
  </label>
  <div
    class="c5"
  >
    <label
      aria-hidden="true"
      class="c6 c7"
      disabled=""
      for=":ro:"
      id=":rp:"
    >
      Unchecked
    </label>
  </div>
</div>
`;

exports[`render Checkbox disabled unchecked 1`] = `
.c6 {
  color: #313440;
  margin: 0 0 1rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5rem;
}

.c6:last-child {
  margin-bottom: 0;
}

.c7 {
  cursor: pointer;
  cursor: not-allowed;
}

.c5 {
  margin-left: 0.5rem;
}

.c0 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c4 {
  -webkit-transition: all 150ms ease-out;
  transition: all 150ms ease-out;
  -webkit-transition-property: border-color,background,box-shadow,color,opacity;
  transition-property: border-color,background,box-shadow,color,opacity;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #FFFFFF;
  box-sizing: border-box;
  border: 1px solid #D9DCE9;
  border-color: #D9DCE9;
  border-radius: 0.25rem;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 1.25rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 0.125rem;
  margin-top: 0.125rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1.25rem;
  background: #F6F7FC;
  border-color: #D9DCE9;
  cursor: not-allowed;
}

.c1:focus + .c3 {
  box-shadow: 0 0 0 0.25rem #DBE3FE;
}

.c4 svg {
  opacity: 0;
}

<div
  class="c0"
>
  <input
    aria-checked="false"
    aria-labelledby=":rm:"
    class="c1 c2"
    disabled=""
    id=":rl:"
    type="checkbox"
  />
  <label
    aria-hidden="true"
    class="c3 c4"
    disabled=""
    for=":rl:"
  >
    <svg
      aria-hidden="true"
      class="brjSIf"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M9 16.17 5.53 12.7a.996.996 0 1 0-1.41 1.41l4.18 4.18c.39.39 1.02.39 1.41 0L20.29 7.71a.996.996 0 1 0-1.41-1.41z"
      />
    </svg>
  </label>
  <div
    class="c5"
  >
    <label
      aria-hidden="true"
      class="c6 c7"
      disabled=""
      for=":rl:"
      id=":rm:"
    >
      Checked
    </label>
  </div>
</div>
`;

exports[`render Checkbox indeterminate 1`] = `
.c6 {
  color: #313440;
  margin: 0 0 1rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5rem;
}

.c6:last-child {
  margin-bottom: 0;
}

.c7 {
  cursor: pointer;
}

.c5 {
  margin-left: 0.5rem;
}

.c0 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c4 {
  -webkit-transition: all 150ms ease-out;
  transition: all 150ms ease-out;
  -webkit-transition-property: border-color,background,box-shadow,color,opacity;
  transition-property: border-color,background,box-shadow,color,opacity;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #3C64F4;
  box-sizing: border-box;
  border: 1px solid #D9DCE9;
  border-color: #3C64F4;
  border-radius: 0.25rem;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 1.25rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 0.125rem;
  margin-top: 0.125rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1.25rem;
}

.c4:hover {
  border-color: #3C64F4;
}

.c1:focus + .c3 {
  box-shadow: 0 0 0 0.25rem #DBE3FE;
}

.c4 svg {
  opacity: 1;
}

<div
  class="c0"
>
  <input
    aria-labelledby=":rg:"
    class="c1 c2"
    id=":rf:"
    type="checkbox"
  />
  <label
    aria-hidden="true"
    class="c3 c4"
    for=":rf:"
  >
    <svg
      aria-hidden="true"
      class="brjSIf"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M18 13H6c-.55 0-1-.45-1-1s.45-1 1-1h12c.55 0 1 .45 1 1s-.45 1-1 1"
      />
    </svg>
  </label>
  <div
    class="c5"
  >
    <label
      class="c6 c7"
      for=":rf:"
      id=":rg:"
    >
      Unchecked
    </label>
  </div>
</div>
`;

exports[`render Checkbox unchecked 1`] = `
.c6 {
  color: #313440;
  margin: 0 0 1rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5rem;
}

.c6:last-child {
  margin-bottom: 0;
}

.c7 {
  cursor: pointer;
}

.c5 {
  margin-left: 0.5rem;
}

.c0 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c4 {
  -webkit-transition: all 150ms ease-out;
  transition: all 150ms ease-out;
  -webkit-transition-property: border-color,background,box-shadow,color,opacity;
  transition-property: border-color,background,box-shadow,color,opacity;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #FFFFFF;
  box-sizing: border-box;
  border: 1px solid #D9DCE9;
  border-color: #D9DCE9;
  border-radius: 0.25rem;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 1.25rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 0.125rem;
  margin-top: 0.125rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1.25rem;
}

.c4:hover {
  border-color: #B4BAD1;
}

.c1:focus + .c3 {
  box-shadow: 0 0 0 0.25rem #DBE3FE;
}

.c4 svg {
  opacity: 0;
}

<div
  class="c0"
>
  <input
    aria-checked="false"
    aria-labelledby=":r4:"
    class="c1 c2"
    id=":r3:"
    type="checkbox"
  />
  <label
    aria-hidden="true"
    class="c3 c4"
    for=":r3:"
  >
    <svg
      aria-hidden="true"
      class="brjSIf"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M9 16.17 5.53 12.7a.996.996 0 1 0-1.41 1.41l4.18 4.18c.39.39 1.02.39 1.41 0L20.29 7.71a.996.996 0 1 0-1.41-1.41z"
      />
    </svg>
  </label>
  <div
    class="c5"
  >
    <label
      class="c6 c7"
      for=":r3:"
      id=":r4:"
    >
      Unchecked
    </label>
  </div>
</div>
`;

exports[`render Checkbox with description object 1`] = `
.c6 {
  color: #313440;
  margin: 0 0 1rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5rem;
}

.c6:last-child {
  margin-bottom: 0;
}

.c8 {
  color: #313440;
  margin: 0 0 1rem;
  color: #5E637A;
  font-size: 0.875rem;
  font-weight: 400;
  line-height: 1.25rem;
  margin: 0 0 0.75rem;
}

.c8:last-child {
  margin-bottom: 0;
}

.c7 {
  cursor: pointer;
}

.c9 {
  -webkit-transition: all 70ms ease-out;
  transition: all 70ms ease-out;
  -webkit-transition-property: color;
  transition-property: color;
  color: #3C64F4;
  cursor: pointer;
  font-size: 1rem;
  font-weight: 400;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c9:active {
  color: #0024A6;
}

.c9:hover:not(:active) {
  color: #0024A6;
}

.c10 {
  font-size: 0.875rem;
}

.c5 {
  margin-left: 0.5rem;
}

.c0 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c4 {
  -webkit-transition: all 150ms ease-out;
  transition: all 150ms ease-out;
  -webkit-transition-property: border-color,background,box-shadow,color,opacity;
  transition-property: border-color,background,box-shadow,color,opacity;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #FFFFFF;
  box-sizing: border-box;
  border: 1px solid #D9DCE9;
  border-color: #D9DCE9;
  border-radius: 0.25rem;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 1.25rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 0.125rem;
  margin-top: 0.125rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1.25rem;
}

.c4:hover {
  border-color: #B4BAD1;
}

.c1:focus + .c3 {
  box-shadow: 0 0 0 0.25rem #DBE3FE;
}

.c4 svg {
  opacity: 0;
}

<div
  class="c0"
>
  <input
    aria-checked="false"
    aria-labelledby=":r7:"
    class="c1 c2"
    id=":r6:"
    name="test-group"
    type="checkbox"
  />
  <label
    aria-hidden="true"
    class="c3 c4"
    for=":r6:"
  >
    <svg
      aria-hidden="true"
      class="brjSIf"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M9 16.17 5.53 12.7a.996.996 0 1 0-1.41 1.41l4.18 4.18c.39.39 1.02.39 1.41 0L20.29 7.71a.996.996 0 1 0-1.41-1.41z"
      />
    </svg>
  </label>
  <div
    class="c5"
  >
    <label
      class="c6 c7"
      for=":r6:"
      id=":r7:"
    >
      Unchecked
    </label>
    <p
      class="c8"
    >
      description
       
      <a
        class="c9 c10"
        href="bar"
        target="foo"
      >
        learn more
      </a>
    </p>
  </div>
</div>
`;

exports[`render Checkbox with description string 1`] = `
.c6 {
  color: #313440;
  margin: 0 0 1rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5rem;
}

.c6:last-child {
  margin-bottom: 0;
}

.c8 {
  color: #313440;
  margin: 0 0 1rem;
  color: #5E637A;
  font-size: 0.875rem;
  font-weight: 400;
  line-height: 1.25rem;
  margin: 0 0 0.75rem;
}

.c8:last-child {
  margin-bottom: 0;
}

.c7 {
  cursor: pointer;
}

.c5 {
  margin-left: 0.5rem;
}

.c0 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c4 {
  -webkit-transition: all 150ms ease-out;
  transition: all 150ms ease-out;
  -webkit-transition-property: border-color,background,box-shadow,color,opacity;
  transition-property: border-color,background,box-shadow,color,opacity;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #FFFFFF;
  box-sizing: border-box;
  border: 1px solid #D9DCE9;
  border-color: #D9DCE9;
  border-radius: 0.25rem;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 1.25rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 0.125rem;
  margin-top: 0.125rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1.25rem;
}

.c4:hover {
  border-color: #B4BAD1;
}

.c1:focus + .c3 {
  box-shadow: 0 0 0 0.25rem #DBE3FE;
}

.c4 svg {
  opacity: 0;
}

<div
  class="c0"
>
  <input
    aria-checked="false"
    aria-labelledby=":rd:"
    class="c1 c2"
    id=":rc:"
    name="test-group"
    type="checkbox"
  />
  <label
    aria-hidden="true"
    class="c3 c4"
    for=":rc:"
  >
    <svg
      aria-hidden="true"
      class="brjSIf"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M9 16.17 5.53 12.7a.996.996 0 1 0-1.41 1.41l4.18 4.18c.39.39 1.02.39 1.41 0L20.29 7.71a.996.996 0 1 0-1.41-1.41z"
      />
    </svg>
  </label>
  <div
    class="c5"
  >
    <label
      class="c6 c7"
      for=":rc:"
      id=":rd:"
    >
      Unchecked
    </label>
    <p
      class="c8"
    >
      description text
    </p>
  </div>
</div>
`;
